raft/raft.go

```diff
@@ -52,6 +52,12 @@ func (st StateType) String() string {
 type Progress struct {
 	Match, Next uint64
 	Wait        int
+	// If the last MsgAppResp updates the Match, which indicates that the
+	// follower has caught up the progress, CatchUp will be set.
+	// If a valid MsgApp with rejection is received, CatchUp will be unset.
+	// When CatchUp is set, leader can send MsgApp optimistically to the
+	// follower.
+	CatchUp bool
```

Comment:
how about just Matched? The previous problem is that "Match!=0" is not a good indication of a Match.